src/icrar/leap-accelerate/algorithm/cpu/PhaseRotate.cc

```diff
@@ -55,53 +57,129 @@ namespace icrar
 {
 namespace cpu
 {
-    void RemoteCalibration(cuda::MetaData& metadata, const Eigen::Matrix<casacore::MVDirection, Eigen::Dynamic, 1>& directions)
+    CalibrateResult Calibrate(
+        const icrar::MeasurementSet& ms,
+        const std::vector<icrar::MVDirection>& directions,
+        int solutionInterval)
     {
 
+        auto metadata = icrar::casalib::MetaData(ms);
+
+        auto output_integrations = std::vector<std::queue<cpu::IntegrationResult>>();
+        auto output_calibrations = std::vector<std::queue<cpu::CalibrationResult>>();
+        auto input_queues = std::vector<std::vector<cpu::Integration>>();
+        
+        for(int i = 0; i < directions.size(); ++i)
+        {
+            auto queue = std::vector<cpu::Integration>(); 
+            queue.push_back(cpu::Integration(
+                ms,
+                i,
+                metadata.channels,
+                metadata.GetBaselines(),
+                metadata.num_pols));
+
+            input_queues.push_back(queue);
+            output_integrations.push_back(std::queue<cpu::IntegrationResult>());
+            output_calibrations.push_back(std::queue<cpu::CalibrationResult>());
+        }
+
+        for(int i = 0; i < directions.size(); ++i)
+        {
+            metadata.SetDD(directions[i]);
+            metadata.SetWv();
+            metadata.avg_data = casacore::Matrix<DComplex>(metadata.GetBaselines(), metadata.num_pols);
+
+            auto metadatahost = icrar::cpu::MetaData(metadata); // use other constructor
+            icrar::cpu::PhaseRotate(metadatahost, directions[i], input_queues[i], output_integrations[i], output_calibrations[i]);
+        }
+
+        return std::make_pair(std::move(output_integrations), std::move(output_calibrations));
     }
 
     void PhaseRotate(
-        cuda::MetaData& metadata,
-        const casacore::MVDirection& directions,
-        std::queue<Integration>& input,
-        std::queue<IntegrationResult>& output_integrations,
-        std::queue<CalibrationResult>& output_calibrations)
+        cpu::MetaData& metadata,
+        const icrar::MVDirection& direction,
+        std::vector<cpu::Integration>& input,
+        std::queue<cpu::IntegrationResult>& output_integrations,
+        std::queue<cpu::CalibrationResult>& output_calibrations)
```

Comment:
You already are in the `icrar::cpu` namespace but there are many `cpu::`-prefixed names still, which seems unnecessary. I highlighted this change because I realized here, but I can see a few more below at least, maybe there are more.

Reply:
This is mostly leftover as some of these classes were/are also in the icrar namespace. Currently I still have a casalib::MetaData class floating around in here which still needs to be upgraded once I verify the behaviour is identical to the casalib port